server/src/com/thoughtworks/go/server/service/JobInstanceService.java

```diff
@@ -133,7 +126,7 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
                 internalUpdateJobStateAndResult(job);
                 /* POST JOB COMPLETION HOOK - BEGIN */
```

Comment:
I guess we could get rid of the `if` block and the comments around it. Looks good otherwise.